astropy/io/ascii/src/tokenizer.c

```diff
@@ -595,7 +596,7 @@ long str_to_long(tokenizer_t *self, char *str)
     char *tmp;
     long ret;
     errno = 0;
-    ret = strtol(str, &tmp, 0);
+    ret = strtol(str, &tmp, 10);
```

Comment:
A possible alternative would be to provide `base` as an additional keyword arg, but then it should better be supported by the slow reader as well. Not that far out of reach, since Python/Numpy `int()` does accept the same `base` option, unfortunately I don't see an easy way to get it handed down through the `np.array` call that handles the conversion.
Probably out of scope for the current PR.